### PR TITLE
Minor code cleanup

### PR DESCRIPTION
**Pull request by ninjalulz**
Related to #1809 & #1883 

Special thanks to @ranisalt

### CODE REVIEW

File: src/player.cpp [hunk 1]
```diff
@@ -1990,33 +1990,24 @@ void Player::death(Creature* lastHitCreature)
...
-				for (const auto& it : damageMap) {
-					CountBlock_t cb = it.second;
-					if ((OTSYS_TIME() - cb.ticks) <= inFightTicks) {
-						Player* damageDealer = g_game.getPlayerByID(it.first);
-						if (damageDealer) {
-							sumLevels += damageDealer->getLevel();
-						}
+		bool lastHitPlayer = lastHitIsPlayer(lastHitCreature);
+
+		if (lastHitPlayer) {
```
Comment (ranisalt) [inline src/player.cpp]: Can shrink to `if (lastHitIsPlayer(lastHitCreature))`.

File: src/player.h [hunk 2]
```diff
@@ -594,6 +594,7 @@ class Player final : public Creature, public Cylinder
 		bool isImmune(ConditionType_t type) const final;
 		bool hasShield() const;
 		bool isAttackable() const final;
+		bool lastHitIsPlayer(Creature* lastHitCreature) const;
```
Comment (ranisalt) [inline src/player.h]: Use a free function, not a method for faster compiling as it's only useful in player.cpp compilation unit (will never be used outside)

File: src/player.cpp (hunk 1 above)
Comment (gugahoa) [inline src/player.cpp]: I think he did that way because of what's at line 2107

Comment (ranisalt) [inline src/player.cpp]: Yes, I'm short sighted and wasn't wearing my internet glasses.

File: src/player.h (hunk 2 above)
Comment (ninjalulz) [inline src/player.h]: Alright. I have never heard of the expression 'free function' before, are you simply referring to a non-member function that you only declare inside the cpp file? If so, what is the preferred method to do so? Declare the function before the member function where it's being utilized, or at the bottom (and provide a function prototype for the function at the top).

Comment (raymondtfr) [inline src/player.h]: I guess  @ranisalt meant a "lambda"? or a ``friend`` function. :dancers:

Comment (marksamman) [inline src/player.h]: I think he means moving the function into player.cpp and not declaring it in the header file. You could also just make it static.

Comment (ninjalulz) [inline src/player.h]: I see, well I decided to make it static instead to avoid any code inconsistency. :smile:

**Comment by marksamman** [comment]
Thanks!

File: src/player.h (hunk 2 above)
Comment (ranisalt) [inline src/player.h]: Yes, you understood correctly. A free function is a non-member function, and if you make it static and only on the cpp it's not visible outside that file compilation unit, so you don't need to recompile if the function signature ever changes, etc.

Declare it just before the function it is used. A static lambda is also useful if it is used only by a single method.

**Pull request merged**